core/src/main/java/org/elasticsearch/index/translog/TranslogWriter.java

```diff
@@ -313,14 +268,9 @@ public boolean syncUpTo(long offset) throws IOException {
 
     @Override
     protected void readBytes(ByteBuffer targetBuffer, long position) throws IOException {
-        try (ReleasableLock lock = readLock.acquire()) {
-            if (position >= writtenOffset) {
-                assert targetBuffer.hasArray() : "buffer must have array";
-                final int sourcePosition = (int) (position - writtenOffset);
-                System.arraycopy(buffer, sourcePosition,
-                    targetBuffer.array(), targetBuffer.position(), targetBuffer.limit());
-                targetBuffer.position(targetBuffer.limit());
-                return;
+        if (position+targetBuffer.limit() > getWrittenOffset()) {
```

Comment:
Nit: strictly speaking i think we need targetBuffer.remaining()  , which is how many bytes we are reading.